File: codeforces/cf1401C_mere_array.cpp
// Garo9521
// https://codeforces.com/contest/1401/problem/C
// Math

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

void solve() {
    int N;
    cin >> N;
    vector <int> A(N), sorted(N);
    for (int i = 0; i < N; ++i) {
        cin >> A[i]; 
        sorted[i] = A[i];
    }
    sort(A.begin(), A.end());
    int minimum_number = A[0];
    for (int i = 0; i < N; ++i) {
        if(sorted[i] == A[i] || __gcd(A[i], minimum_number) == minimum_number)
            continue;
        cout << "NO\n";
        return;
    }
    cout << "YES\n"; 
}

int main() {
    int T;
    cin >> T;
    while (T--) {
       solve();
    }
    return 0;
}

File: codeforces/cf1401C_mere_array.rs
// uninhm
// https://codeforces.com/contest/1401/problem/C
// math

use std::io::stdin;

fn main() {
    let mut t = String::new();
    stdin().read_line(&mut t).unwrap();
    let t: i64 = t.trim().parse().unwrap();

    for _ in 0 .. t {
        let mut n = String::new();
        stdin().read_line(&mut n).unwrap();
        let n: i64 = n.trim().parse().unwrap();

        let mut line = String::new();
        stdin().read_line(&mut line).unwrap();

        let a: Vec<i64> = line
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();

        let mut b = a.clone();

        b.sort();

        let m = b[0];

        let mut flag = true;
        for i in 0 .. n as usize {
            if a[i] != b[i] && a[i] % m != 0 {
                flag = false;
                break;
            }
        }
        println!("{}",
            if flag {
                "YES"
            } else {
                "NO"
            }
        );
    }
}
